src/executor/executor.cc

```diff
@@ -0,0 +1,66 @@
+#include "executor/executor.hh"
+
+namespace sim {
+
+static void executeAdd(const Instruction &inst, State &state) {
+  RegVal rs1 = state.regs.get(inst.rs1);
+  RegVal rs2 = state.regs.get(inst.rs2);
+  state.regs.set(inst.rd, rs1 + rs2);
```

Comment:
I think we should change the behavior of `State::set()` - it should do nothing in case of setting to `x0` register, because, for example, `NOP` instruction looks like `ADD x0, x0, 0` - assignment to `x0`, furthermore `JALR` supports `x0` as a destination register (to store return address if it is not needed). 
I will create issue on this

@Tako-San, @Bryanskiy – who can change it ASAP?